astropy/nddata/utils.py

```diff
@@ -104,14 +115,23 @@ def overlap_slices(large_array_shape, small_array_shape, position, mode="partial
             '"position" must have the same number of dimensions as "small_array_shape".'
         )
 
+    if not callable(limit_rounding_method):
+        raise ValueError("Limit rounding method must be a callable function.")
+
     # define the min/max pixel indices
-    indices_min = [
-        int(np.ceil(pos - (small_shape / 2.0)))
-        for (pos, small_shape) in zip(position, small_array_shape)
-    ]
+    # round according to the limit_rounding_method
+    try:
+        indices_min = [
+            int(limit_rounding_method(pos - (small_shape / 2.0)))
+            for (pos, small_shape) in zip(position, small_array_shape)
+        ]
+    except (TypeError, ValueError):
+        raise ValueError(
+            "Limit rounding method must accept a single number as input and return a single number."
+        )
```

Comment:
let's chain the exceptions is a logical fashion, so it doesn't look like we crashed while handling
```suggestion
    except (TypeError, ValueError) as exc:
        raise ValueError(
            "Limit rounding method must accept a single number as input and return a single number."
        ) from exc
```